openapi-validator/src/test/java/org/ballerinalang/openapi/validator/tests/ResourceToOperationTests.java

```diff
@@ -0,0 +1,182 @@
+        Assert.assertEquals(((TypeMismatch) (validationError).get(0)).getTypeJsonSchema(), Constants.Type.STRING);
+        Assert.assertEquals(((TypeMismatch) (validationError).get(0)).getTypeBallerinaType(), Constants.Type.INT);
+    }
+    //paramName change
+    @Test(description = "invalid test for Path parameter: parameter miss -  ")
+    public void testMissingPathParam() throws OpenApiValidatorException, IOException {
+        Path contractPath = RES_DIR.resolve("swagger/invalid/petstore_path_parameter.yaml");
+        api = ServiceValidator.parseOpenAPIFile(contractPath.toString());
+        project = ValidatorTest.getProject(RES_DIR.resolve("ballerina/invalid/petstore_path_parameter.bal"));
+        operation = api.getPaths().get("/pets/{petId}").getGet();
+        resourceMethod = ValidatorTest.getResourceMethod(project, "/pets/{petId}", "get");
+        syntaxTree = ValidatorTest.getSyntaxTree(project);
+        semanticModel = ValidatorTest.getSemanticModel(project);
+        validationError = ResourceValidator.validateResourceAgainstOperation(operation, resourceMethod, semanticModel
+                , syntaxTree);
+        Assert.assertTrue(validationError.get(0) instanceof TypeMismatch);
+        Assert.assertEquals(validationError.get(0).getFieldName(), "petId");
+        Assert.assertEquals(((TypeMismatch) (validationError).get(0)).getTypeJsonSchema(), Constants.Type.STRING);
+        Assert.assertEquals(((TypeMismatch) (validationError).get(0)).getTypeBallerinaType(), Constants.Type.INT);
+    }
+
+    @Test(description = "invalid test for Path parameter: when path has two path parameters")
+    public void testTypeMissMatchTwoPathParam() throws OpenApiValidatorException, IOException {
+        Path contractPath = RES_DIR.resolve("swagger/invalid/petstore_path_2parameter.yaml");
+        api = ServiceValidator.parseOpenAPIFile(contractPath.toString());
+        project = ValidatorTest.getProject(RES_DIR.resolve("ballerina/invalid/petstore_path_2parameter.bal"));
+        operation = api.getPaths().get("/pets/{petId}/{petName}").getGet();
+        resourceMethod = ValidatorTest.getResourceMethod(project, "/pets/{petId}/{petName}", "get");
+        syntaxTree = ValidatorTest.getSyntaxTree(project);
+        semanticModel = ValidatorTest.getSemanticModel(project);
+        validationError = ResourceValidator.validateResourceAgainstOperation(operation, resourceMethod, semanticModel
+                , syntaxTree);
+        Assert.assertTrue(validationError.get(0) instanceof TypeMismatch);
+        Assert.assertEquals(validationError.get(0).getFieldName(), "petId");
+        Assert.assertEquals(((TypeMismatch) (validationError).get(0)).getTypeJsonSchema(), Constants.Type.STRING);
+        Assert.assertEquals(((TypeMismatch) (validationError).get(0)).getTypeBallerinaType(), Constants.Type.INT);
+    }
+//    'int', 'string', 'float', 'boolean', 'decimal'
+    //Query param -
```

Comment:
Format error, Also shall we add those comments to the description if we need them?